src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -967,6 +967,15 @@ winrt::fire_and_forget AppHost::_setupGlobalHotkeys()
     // The hotkey MUST be registered on the main thread. It will fail otherwise!
     co_await winrt::resume_foreground(_logic.GetRoot().Dispatcher());
 
+    if (!_window)

```

Comment:
Ah we null it out properly too? Awesome

Reply:
Wait, is this sufficient? The only time we will observe `_window` to be null is _after our destructor has run_... which would make accessing `&this->_window` an AV (or UB). We need to make sure that AppHost is properly lifetime-managed _before we get here_

Reply:
AH YOU'D THINK

Looking at the stack, this hits in this sliver of time where we've called `~AppHost`, we've nulled out the `_window`, but then the Dispatcher like, runs a bunch more and... flushes callbacks? I'm not sure what it's doing here. There's been miscellaneous crashes over the last few months where like, we try to set the title of the window but it _doesn't exist_, because we're in this bit of Dispatcher rundown. This is just another version of that. They're all crashes once we've already started tearing down, so they're not that user-impactful, but they still happen.

IDK if it's safe to just gank the `Dispatcher` before we null out the `_window` to release the XAML bits.

Reply:
If the `~AppHost` dtor is not on the stack, it is still UB to access this member; I'm concerned this is a band-aid? :)

Reply:
Oh sorry, to clarify, the `~AppHost` dtor _is_ on the stack here.